sdks/java/harness/src/test/java/org/apache/beam/fn/harness/state/FakeBeamFnStateClient.java

```diff
@@ -93,8 +118,18 @@ public void handle(
         break;
 
       case APPEND:
-        ByteString previousValue = data.getOrDefault(request.getStateKey(), ByteString.EMPTY);
-        data.put(request.getStateKey(), previousValue.concat(request.getAppend().getData()));
+        List<ByteString> previousValue =
+            data.getOrDefault(request.getStateKey(), Collections.singletonList(ByteString.EMPTY));
+        List<ByteString> newValue = new ArrayList<>();
+        newValue.addAll(previousValue);
+        ByteString newData = request.getAppend().getData();
+        if (newData.size() % 2 == 0) {
+          newValue.remove(newValue.size() - 1);
+          newValue.add(previousValue.get(previousValue.size() - 1).concat(newData));
+        } else {
+          newValue.add(newData);
+        }
```

Comment:
Just curious why it's `newData.size() % 2 == 0`?

Reply:
This was just to (deterministically) append about half the time and add a new item about half the time for testing.